modules/wallet/database.go

```diff
@@ -206,6 +207,39 @@ func dbDeleteSpentOutput(tx *bolt.Tx, id types.OutputID) error {
 	return dbDelete(tx.Bucket(bucketSpentOutputs), id)
 }
 
+// dbReinitialize wipes and reinitializes a wallet database.
+func dbReinitialize(tx *bolt.Tx) error {
+	for _, bucket := range dbBuckets {
+		err := tx.DeleteBucket(bucket)
+		if err != nil {
+			return err
+		}
+		_, err = tx.CreateBucket(bucket)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
That's why I want to see this tested out. The wallet db is 1 GB, if Bolt doesn't efficiently reuse the space then users doing multiple re-inits could end up with several extra GB on disk.

It's easy enough to test.